libs/island-ui/core/src/lib/Select/Components/index.tsx

```diff
@@ -23,11 +23,31 @@ export const Menu = (props: MenuProps<ReactSelectOption>) => (
 
 export const Option = (props: OptionProps<ReactSelectOption>) => {
   const size: SelectProps['size'] = props.selectProps.size || 'md'
+  const description = props.data?.description
+  // Truncate description by default
+  const descriptionTruncated =
+    !!description && props.data?.descriptionTruncate !== false
+
   return (
     <components.Option
       className={cn(styles.option, styles.optionSizes[size!])}
       {...props}
-    />
+    >
+      <>
+        {props.children}
+        {!!description && (
+          <div
+            className={cn(
+              styles.optionDescription,
+              styles.optionDescriptionSizes[size!],
```

Comment:
do you need the non-null assertion operator here?
```suggestion
              styles.optionDescriptionSizes[size],
```